lib/cloudregister/registerutils.py

```diff
@@ -2226,10 +2230,17 @@ def remove_registration_data():
             scc_except_msg += 'artifacts removed locally.'
             logging.error(scc_except_msg % user)
             logging.warning(e)
+        logging.info('Deleting locally stored credentials')
+        __remove_credentials('suse.com')
+        credentials_cleaned = True
         __remove_repo_artifacts('suse.com')
     else:
         logging.info('No current registration server set.')
 
+    if not credentials_cleaned:
+        logging.info('Deleting locally stored credentials')
+        __remove_credentials('')
```

Comment:
Can we use the key value caller semantics

```
__remove_credentials(smt_server_name=some_value)
```

I'm asking because I think with an empty string as value the call is hard to read and I guess a None value like with __remove_credentials() is causing other issues.

Thoughts ?

Reply:
It's all nitpick for readability. I leave it up to you if you want that or not.

It looks good to me either way

Reply:
I do not think this is going in the direction it should be. This fixes the symptom that has been observed but does not explain why we leave stuff behind in the current implementation in the first place. The lack of explanation of "circumstances" in the commit also indicates that this is based a band aid to the currently observed symptoms.

Reply:
I am going to dig into this in more detail.